pypeit/core/coadd1d.py

```diff
@@ -791,13 +807,9 @@ def sensfunc_weights(sensfile, waves, debug=False):
     Returns:
         ndarray: sensfunc weights evaluated on the input waves
         wavelength grid
-    '''
-
-    sens_meta= Table.read(sensfile, 1)
-    sens_table = Table.read(sensfile, 2)
-    func = sens_meta['FUNC'][0]
-    polyorder_vec = sens_meta['POLYORDER_VEC'][0]
+    """
 
+    wave_sens, sens, meta_table, out_table, header_sens = sensfunc.SensFunc.load(sensfile)
```

Comment:
Once the data container is finished we can make a sensfunc object and pass that back. I did it this way, partly because the meta_table and out_table are astropy tables which have different information depending on which algorithm was used for the sensfunc, and there are also headaches associated with splicing. But yes in principle these quantities could be loaded into attributes of an object.